docker/docker-compose.yml

```diff
@@ -0,0 +1,178 @@
+version: "3.8"
+name: hskauting
+
+networks:
+    traefik:
+
+volumes:
+    mysql:
+    mysql-test:
+
+x-php-base:
+    &php-base
+    build:
+        context: ./containers/php
+    working_dir: /app
+    env_file:
+        - ./containers/php/ini/image.ini
+    networks:
+        - traefik
+    volumes:
+        - ../:/app
+        - ./containers/php/ini/php.ini:/etc/php/8.1/fpm/conf.d/php.ini
+        - ./containers/php/ini/php.ini:/etc/php/8.1/cli/conf.d/php.ini
+
+x-mysql-base:
+    &mysql-base
+    image: mysql:8.0
+    networks:
+        - traefik
+    environment:
+        MYSQL_DATABASE: hskauting
+        MYSQL_USER: hskauting
+        MYSQL_PASSWORD: hskauting
+        MYSQL_ROOT_PASSWORD: root
+        MYSQL_COLLATION: utf8mb3_czech_ci
+        MYSQL_CHARACTER_SET: utf8
+
+services:
+    traefik:
+        image: traefik:latest
+        restart: unless-stopped
+        networks:
+            - traefik
+        ports:
+            - "80:80"
+            - "3000:3000"
+        volumes:
+            - /var/run/docker.sock:/var/run/docker.sock:ro
+            - ./containers/traefik/traefik.yml:/traefik.yml:ro
+        labels:
+            traefik.enable: true
+            traefik.http.routers.api.rule: Host(`traefik.localhost`)
+            traefik.http.routers.api.entrypoints: web
+            traefik.http.routers.api.service: api@internal
+
+    php:
+        <<: *php-base
+        container_name: hskauting.app #Backward Compatibility
+        env_file:
+            - containers/php/ini/image.ini
+            - ./../.env.local
+        depends_on:
+            - mysql
+        labels:
+            traefik.enable: true
+            traefik.http.routers.php.rule: Host(`vite.localhost`)
+            traefik.http.routers.php.entrypoints: web, assets
+            traefik.http.services.php.loadbalancer.server.port: 3000
+
+    php-xdebug:
+        <<: *php-base
+        hostname: php-xdebug.localhost
+        depends_on:
+            - mysql
+        env_file:
+            - containers/php/ini/image.ini
+            - containers/php/ini/xdebug.ini
+            - ./../.env.local
+    php-test:
+        <<: *php-base
+        hostname: php-test.localhost
+        container_name: hskauting.app-test
+        depends_on:
+            - mysql-test
+        env_file:
+            - containers/php/ini/image.ini
+            - containers/php/ini/xdebug.ini
+            - ./../.env.test
+
+    nginx:
```

Comment:
proc nginx kdyz tu mas i traefik? na lebede bezi apache

Reply:
Nejsem si úplně jistý, jestli traefik podporuje FastCGI protocol. V minulosti to bez nginx nešlo